script/test-utils.ts

```diff
@@ -0,0 +1,53 @@
+import path from 'node:path';
+import fs from 'node:fs';
+import { tmpdir } from 'node:os';
+import { execSync } from 'node:child_process';
+
+export const PNPM_STORE_PATH = path.resolve(__dirname, '../.pnpm-test-store');
+export const NPM_RC_FILE = '.npmrc';
+export const NPM_RC_CONTENTS = `store-dir = ${PNPM_STORE_PATH}`;
+export const PACKAGE_JSON_FILE = 'package.json';
+export const PACKAGE_JSON_CONTENTS = '{"name":"test-project","version":"1.0.0"}';
+
+export function preparePackageJson(dependencies: {[key: string]: string} = {}, devDependencies: {[key: string]: string} = {}): string {
+  const tmpDir = fs.mkdtempSync(path.join(tmpdir(), 'zenstack-test-'));
+  try {
+  const packageJsonContents = 
+`{
+  "name":"test-project",
+  "version":"1.0.0",
+  "dependencies": {
+    ${Object.entries(dependencies).map(([k, v]) => `"${k}": "${v}"`).join(',\n')}
+  },
+  "devDependencies": {
+    ${Object.entries(devDependencies).map(([k, v]) => `"${k}": "${v}"`).join(',\n')}
+  }
+}`;
+
+  initProjectDir(tmpDir, packageJsonContents, false);
```

Comment:
I considered doing a `pnpm store add` here instead of a plain install. While that worked, I decided against it in the end because it's a secondary way of processing the dependencies and I didn't see a significant downside to just installing and throwing the local project away right after.